config/initializers/sidekiq.rb

```diff
@@ -18,6 +19,7 @@
   # middleware to spec/rails_helper.rb to run in tests as well
   config.server_middleware do |chain|
     chain.add WorkerHealthChecker::Middleware
+    chain.add NoRetryJobs
```

Comment:
Would it be easier to add retry config to each job class, rather than all jobs? We may want retryable jobs in the future? 

https://github.com/mperham/sidekiq/wiki/Error-Handling#configuration

Or we could confgure our base class to have this?

```
  sidekiq_options :retry => false
```

Reply:
ActiveJob does not support `sidekiq_options`: https://github.com/mperham/sidekiq/wiki/Active-Job

I spent quite some time researching this and trying different things, and as far as I can tell, this is the only way to set the retry to false and also raise the error. ActiveJob has a [discard_on](http://api.rubyonrails.org/classes/ActiveJob/Exceptions/ClassMethods.html) option that can be applied to a particular job and a particular error, which will stop the job from retrying, but it won't raise the error, so we'll never know about it.

However, what we can do, since we have access to the queue in `NoRetryJobs`, is to add a conditional to only apply this to the `idv` queue for now (I will be submitting another PR to apply this to email, sms, and voice as well). I just tested it and it works. I will push the change and add the missing specs.